src/library/library/library.service.ts

```diff
@@ -72,10 +77,29 @@ export class LibraryService {
     }
 
     const innovationPack =
-      await this.innovationPackService.createInnovationPack(innovationPackData);
+      await this.innovationPackService.createInnovationPack(
+        innovationPackData,
+        library.storageSpace.id
+      );
     library.innovationPacks.push(innovationPack);
     await this.libraryRepository.save(library);
 
     return innovationPack;
   }
+
+  async getStorageSpace(libraryInput: ILibrary): Promise<IStorageSpace> {
+    const library = await this.getLibraryOrFail({
+      relations: ['storageSpace'],
+    });
+    const storageSpace = library.storageSpace;
+
+    if (!storageSpace) {
+      throw new EntityNotFoundException(
+        `Unable to find storage space for Library: ${libraryInput.id}`,
+        LogContext.LIBRARY
```

Comment:
`LogContext` for Storage Space might be better here. The issue is in the library, but is about a Storage Space